apps/shared/avifjpeg.c

```diff
@@ -568,18 +1149,50 @@ avifBool avifJPEGRead(const char * inputFilename,
         }
         avifImageFixXMP(avif); // Remove one trailing null character if any.
     }
+
+#if defined(AVIF_ENABLE_EXPERIMENTAL_JPEG_GAIN_MAP_CONVERSION)
+    // The primary XMP block (for the main image) must contain a node with an hdrgm:Version field if and only if a gain map is present.
+    if (!ignoreGainMap && avifJPEGHasGainMapXMPNode(avif->xmp.data, avif->xmp.size)) {
+        avifJPEGExtractGainMapImage(f, inputFilename, &cinfo, &avif->gainMap, requestedFormat, requestedDepth, chromaDownsampling);
```

Comment:
Returned value is ignored? Should it at least print a warning if the tag is found without a valid gain map?

Reply:
avifJPEGExtractGainMapImage already prints warnings, added a comment.